src/lib/pickles/wrap.mli

```diff
@@ -13,142 +16,118 @@ val wrap :
   -> (   ( Impls.Wrap.Impl.Field.t
          , Impls.Wrap.Impl.Field.t Composition_types.Scalar_challenge.t
          , Impls.Wrap.Impl.Field.t Pickles_types.Shifted_value.Type1.t
-         , ( Impls.Wrap.Impl.Field.t Pickles_types.Shifted_value.Type1.t
-           , Impls.Wrap.Impl.field Snarky_backendless.Cvar.t
-             Snarky_backendless.Snark_intf.Boolean0.t )
-           Pickles_types.Opt.t
          , ( Impls.Wrap.Impl.Field.t Composition_types.Scalar_challenge.t
            , Impls.Wrap.Impl.field Snarky_backendless.Cvar.t
              Snarky_backendless.Snark_intf.Boolean0.t )
            Pickles_types.Opt.t
          , Impls.Wrap.Impl.Boolean.var
-         , Impls.Wrap.Impl.field Snarky_backendless.Cvar.t
-         , Impls.Wrap.Impl.field Snarky_backendless.Cvar.t
-         , Impls.Wrap.Impl.field Snarky_backendless.Cvar.t
-         , ( Impls.Wrap.Impl.field Snarky_backendless.Cvar.t
-             Kimchi_backend_common.Scalar_challenge.t
-             Composition_types.Bulletproof_challenge.t
-           , Pickles_types.Nat.z Backend.Tick.Rounds.plus_n )
-           Pickles_types.Vector.t
-           Pickles_types.Hlist0.Id.t
-         , Impls.Wrap.Impl.field Snarky_backendless.Cvar.t )
-         Import.Types.Wrap.Statement.In_circuit.t
+         , Pasta_bindings.Fq.t Challenge.t
+         , Pasta_bindings.Fq.t Challenge.t
+         , Pasta_bindings.Fq.t Challenge.t
+         , ( Pasta_bindings.Fq.t Challenge.t Kimchi_types.scalar_challenge
+             Bulletproof_challenge.t
+           , Nat.z Backend.Tick.Rounds.plus_n )
+           Vector.vec
+         , Pasta_bindings.Fq.t Challenge.t )
+         Types.Wrap.Statement.In_circuit.t
       -> unit )
   -> typ:('a, 'b) Impls.Step.Typ.t
   -> step_vk:Kimchi_bindings.Protocol.VerifierIndex.Fp.t
-  -> actual_wrap_domains:(Core_kernel.Int.t, 'c) Pickles_types.Vector.t
+  -> actual_wrap_domains:(int, 'c) Vector.vec
   -> step_plonk_indices:'d
   -> feature_flags:Opt.Flag.t Plonk_types.Features.Full.t
   -> actual_feature_flags:bool Plonk_types.Features.t
   -> ?tweak_statement:
-       (   ( Import.Challenge.Constant.t
-           , Import.Challenge.Constant.t Import.Types.Scalar_challenge.t
-           , Backend.Tick.Field.t Pickles_types.Shifted_value.Type1.t
-           , ( Backend.Tick.Field.t Pickles_types.Shifted_value.Type1.t
-             , bool )
-             Import.Types.Opt.t
+       (   ( Challenge.Constant.t
+           , scalar_challenge_constant
+           , Pasta_bindings.Fp.t Shifted_value.Type1.t
            , ( Import.Challenge.Constant.t Import.Types.Scalar_challenge.t
              , bool )
              Import.Types.Opt.t
            , bool
            , 'max_proofs_verified
-             Proof.Base.Messages_for_next_proof_over_same_field.Wrap.t
-           , (int64, Composition_types.Digest.Limbs.n) Pickles_types.Vector.vec
+             Reduced_messages_for_next_proof_over_same_field.Wrap.t
+           , (int64, Nat.N4.n) Vector.vec
```

Comment:
It's more informative to keep the `Digest_limbs.n` here, or just use `Types.Digest.Constant.t`